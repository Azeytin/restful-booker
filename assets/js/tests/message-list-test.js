import React from 'react';
import MessageList from '../src/js/components/MessageList.jsx';
import nock from 'nock';

beforeAll(() => {
    nock('http://localhost')
        .persist()
        .get('/message/')
        .reply(200, {
                messages : [
                    {
                        "name" : "Mark Winteringham",
                        "subject" : "Subject description here",
                        "read" : true
                    }, {
                        "name" : "James Dean",
                        "subject" : "Another description here",
                        "read" : false
                    }, {
                        "name" : "Janet Samson",
                        "subject" : "Lorem ipsum dolores est",
                        "read" : true
                    }
                ]
            }
        );
});

test('Renders the list of messages correctly', async () => {
    const messageComponent = shallow(<MessageList setCount={() => {}} />);

    setTimeout(() => {
        expect(messageComponent).toMatchSnapshot();
    }, 1000);
});

test('Deletes message when selected to delete', (done) => {
    nock('http://localhost')
        .delete('/message/1')
        .reply(201, () => {
            done();
        });

    const messageComponent = shallow(<MessageList setCount={() => {}} />);
    messageComponent.instance().deleteMessage(1);
});

test('Clicking message shows message popup', async () => {
    const messageComponent = shallow(<MessageList setCount={() => {}} />);

    setTimeout(() => {
        messageComponent.find("#message0").simulate('click');
    }, 2000);

    expect(messageComponent).toMatchSnapshot();
});
