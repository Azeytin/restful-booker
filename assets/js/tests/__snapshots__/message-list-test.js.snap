// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clicking message shows message popup 1`] = `
<div>
  <div
    className="messages"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-2 rowHeader"
      >
        <p>
          Name
        </p>
      </div>
      <div
        className="col-sm-9 rowHeader"
      >
        <p>
          Subject
        </p>
      </div>
      <div
        className="col-sm-1"
      />
    </div>
  </div>
</div>
`;
